ui/page/root/receive_page.go

```diff
@@ -168,170 +196,152 @@ func (pg *ReceivePage) generateQRForAddress() {
 	pg.qrImage = &imgdec
 }
 
+func (pg *ReceivePage) getLogo() *cryptomaterial.Image {
+	pg.selectedWallet.GetAssetType()
+	switch pg.selectedWallet.GetAssetType() {
+	case utils.BTCWalletAsset:
+		return pg.Theme.Icons.CircleBTC
+	case utils.DCRWalletAsset:
+		return pg.Theme.Icons.CircleDCR
+	case utils.LTCWalletAsset:
+		return pg.Theme.Icons.CircleLTC
+	default:
+		return nil
+	}
+}
+
 // Layout draws the page UI components into the provided C
 // to be eventually drawn on screen.
 // Part of the load.Page interface.
 func (pg *ReceivePage) Layout(gtx C) D {
-	pg.handleCopyEvent(gtx)
-	pg.pageBackdropLayout(gtx)
-
-	if pg.Load.IsMobileView() {
-		return pg.layoutMobile(gtx)
+	if pg.modalLayout != nil {
+		var modalWidth float32 = 450
+		if pg.IsMobileView() {
+			modalWidth = 0
+		}
+		modalContent := []layout.Widget{pg.contentLayout}
+		return pg.modalLayout.Layout(gtx, modalContent, modalWidth)
 	}
-	return pg.layoutDesktop(gtx)
+	return pg.contentLayout(gtx)
```

Comment:
nit: Consider inverting this if statement. It helps with readability to have smaller blocks in if statements.